### PR TITLE
Nexus 5447 concurrent index remove

### PR DESCRIPTION
**Pull request by ifedorenko**

### CODE REVIEW

File: nexus/plugins/indexer/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/DefaultIndexerManager.java [hunk 1]
```diff
@@ -2505,6 +2550,16 @@ private void exclusiveSingle( Repository repository, Runnable runnable )
      */
     private Lock getRepositoryLock( Repository repository, boolean exclusive )
     {
+        final String lockName = exclusive ? "exclusive" : "shared";
+
+        Thread deleteThread = deleteThreads.get( repository.getId() );
+        if ( deleteThread != null && deleteThread != Thread.currentThread() )
+        {
+            logger.debug( "Could not acquire {} lock on repository {}. The repository {} is being deleted by thread {}.",
+                          lockName, repository.getId(), deleteThread.getName() );
```
Comment (jdillon) [inline nexus/plugins/indexer/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/DefaultIndexerManager.java:2559]: I see 4 {} but only 3 params.

**Comment by cstamas** [comment]
+1

And a totally unrelated but related to introduction of wrapper stream in this pull:
https://github.com/sonatype/nexus/pull/703

**Comment by nabcos** [comment]
+1

**Comment by ifedorenko** [comment]
CI https://bamboo.zion.sonatype.com/browse/NXO-OSSF43-5

**Pull request merged**